lib/internal/crypto/x509.js

```diff
@@ -0,0 +1,334 @@
+'use strict';
+
+const {
+  ObjectSetPrototypeOf,
+  SafeMap,
+  Symbol,
+} = primordials;
+
+const {
+  parseX509,
+  X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT,
+  X509_CHECK_FLAG_NEVER_CHECK_SUBJECT,
+  X509_CHECK_FLAG_NO_WILDCARDS,
+  X509_CHECK_FLAG_NO_PARTIAL_WILDCARDS,
+  X509_CHECK_FLAG_MULTI_LABEL_WILDCARDS,
+  X509_CHECK_FLAG_SINGLE_LABEL_SUBDOMAINS,
+} = internalBinding('crypto');
+
+const {
+  PublicKeyObject,
+  isKeyObject,
+} = require('internal/crypto/keys');
+
+const {
+  customInspectSymbol: kInspect,
+} = require('internal/util');
+
+const {
+  validateBoolean,
+  validateObject,
+  validateString,
+} = require('internal/validators');
+
+const { inspect } = require('internal/util/inspect');
+
+const { Buffer } = require('buffer');
+
+const {
+  isArrayBufferView,
+} = require('internal/util/types');
+
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+    ERR_INVALID_ARG_VALUE,
+  }
+} = require('internal/errors');
+
+const {
+  JSTransferable,
+  kClone,
+  kDeserialize,
+} = require('internal/worker/js_transferable');
```

Comment:
Should [worker_threads.md](https://nodejs.org/api/worker_threads.html#worker_threads_port_postmessage_value_transferlist) get updated to include and link to X509Certificate?

Reply:
Done